app/models/agents/slack_agent.rb

```diff
@@ -67,7 +71,11 @@ def slack_notifier
     def receive(incoming_events)
       incoming_events.each do |event|
         opts = interpolated(event)
-        slack_notifier.ping opts[:message], channel: opts[:channel], username: opts[:username]
+        if /^:/.match(opts[:icon])
+            slack_notifier.ping opts[:message], channel: opts[:channel], username: opts[:username], icon_emoji: opts[:icon]
+        else
```

Comment:
Ah, sorry about the spaces; dang editor defaults.

As for `elsif`, I don't think it's necessary, since Slack will use the default webhook icon if either `icon_emoji` or `icon_url` is empty. Interestingly, it will use a different icon if something invalid is entered, but the webhook still succeeds.